packages/use-codemirror/src/components/CodeMirror/CodeMirror.vue

```diff
@@ -1,44 +1,12 @@
 <script lang="ts" setup>
-import { html } from '@codemirror/lang-html'
-import { java } from '@codemirror/lang-java'
-import { javascript } from '@codemirror/lang-javascript'
-import { json } from '@codemirror/lang-json'
-import { python } from '@codemirror/lang-python'
-import { type LanguageSupport, StreamLanguage } from '@codemirror/language'
-import {
-  c,
-  csharp,
-  kotlin,
-  objectiveC,
-} from '@codemirror/legacy-modes/mode/clike'
-import { clojure } from '@codemirror/legacy-modes/mode/clojure'
-import { go } from '@codemirror/legacy-modes/mode/go'
-import { http } from '@codemirror/legacy-modes/mode/http'
-import { oCaml } from '@codemirror/legacy-modes/mode/mllike'
-import { powerShell } from '@codemirror/legacy-modes/mode/powershell'
-import { r } from '@codemirror/legacy-modes/mode/r'
-import { ruby } from '@codemirror/legacy-modes/mode/ruby'
-import { shell } from '@codemirror/legacy-modes/mode/shell'
-import { swift } from '@codemirror/legacy-modes/mode/swift'
-import * as yamlMode from '@codemirror/legacy-modes/mode/yaml'
-import { type Extension } from '@codemirror/state'
-import {
-  EditorView,
-  type ViewUpdate,
-  keymap,
-  lineNumbers as lineNumbersExtension,
-} from '@codemirror/view'
-import { toRaw, watch } from 'vue'
+import { computed, ref, toRef } from 'vue'
 
 import { useCodeMirror } from '../../hooks'
 import type { CodeMirrorLanguage } from '../../types'
-import { variables } from './extensions/variables'
 
 const props = withDefaults(
   defineProps<{
-    name?: string
-    extensions?: Extension[]
-    content?: string
+    content: string | undefined
```

Comment:
I don’t know what the difference is between those two. Can you enlighten me?

Reply:
The first is an optional property. The second is a required property but the variable provided can be undefined. 

The advantage of the second is a user can't forget to assign a value to it.